src/core/import_species_from_csv.py

```diff
@@ -0,0 +1,170 @@
+# coding: utf8
+
+__contact__ = "info@hytech-imaging.fr"
+__copyright__ = "Copyright (c) 2021 Hytech Imaging"
+
+import csv
+import os.path
+from qgis.core import (
+    QgsVectorLayer,
+    QgsFeature,
+)
+
+
+class ImportSpeciesFromCsv:
+    @staticmethod
+    def importInto(table: QgsVectorLayer):
+        speciesFilePath = os.path.join(
+            os.path.abspath(os.path.dirname(__file__)), "species.csv"
+        )
+        with open(speciesFilePath) as file:
+            lines = file.readlines()
+
+        table.startEditing()
+
+        for i in range(1, len(lines)):
+            params = [
+                '"{}"'.format(x)
+                for x in list(
+                    csv.reader(
+                        [lines[i].strip()], delimiter=",", quotechar='"'
+                    )
+                )[0]
+            ]
+
+            newSpecies = QgsFeature(table.fields())
+            ImportSpeciesFromCsv._setAttributeAsText(
```

Comment:
What do you think about defining a list of parameters and then use a for loop instead:

``` python
columns = ["nom_latin", "type", "Famille", ...]
for index, value in enumerate(columns):
    ImportSpeciesFromCsv._setAttributeAsText(newSpecies, value, params[index])
```

Reply:
The problem is : there are few attributes that are "int", not string.
So it is still possible to use a list but it won't be that elegant.

Reply:
Then I'd probably create 2 lists (one for int columns and another one for str columns) with 2 dedicated loop for.

Reply:
That's what I did in 0c75d5f below

Reply:
Actually you kept a single loop for so you have to do something like `if columnName in columnsOfIntType:`. But it's a minor comment, you can forget that :).